core/controlplane/config/encrypted_assets.go

```diff
@@ -378,31 +406,67 @@ func (r *RawAssetsOnMemory) WriteToDir(dirname string, includeCAKey bool) error
 		{"etcd-key.pem", r.EtcdKey, true},
 		{"etcd-client.pem", r.EtcdClientCert, true},
 		{"etcd-client-key.pem", r.EtcdClientKey, true},
+		{"etcd-trusted-ca.pem", r.EtcdTrustedCA, true},
 		{"kubelet-tls-bootstrap-token", r.TLSBootstrapToken, true},
 
 		// Content entirely provided by user, so do not overwrite it if
 		// the file already exists
 		{"tokens.csv", r.AuthTokens, false},
 	}
+
+	if includeCAKey {
+		assets = append(assets, struct {
+			name      string
+			data      []byte
+			overwrite bool
+		}{"ca-key.pem", r.CACert, true})
+	}
+
 	for _, asset := range assets {
 		path := filepath.Join(dirname, asset.name)
 
-		if asset.name != "ca-key.pem" || includeCAKey {
-			if !asset.overwrite {
-				info, err := os.Stat(path)
-				if info != nil {
-					continue
-				}
-
-				// Unexpected error
-				if err != nil && !os.IsNotExist(err) {
-					return err
-				}
+		if !asset.overwrite {
+			info, err := os.Stat(path)
+			if info != nil {
+				continue
 			}
-			if err := ioutil.WriteFile(path, asset.data, 0600); err != nil {
+
+			// Unexpected error
+			if err != nil && !os.IsNotExist(err) {
 				return err
 			}
 		}
+		if err := ioutil.WriteFile(path, asset.data, 0600); err != nil {
+			return err
+		}
+	}
+
+	// etcd trusted ca and worker-ca are separate files, but pointing to ca.pem by default.
+	// In advanced configurations, when certs are managed outside of kube-aws,
+	// these can be separate CAs to ensure that worker nodes have no certs which would let them
+	// access etcd directly. If worker-ca.pem != ca.pem, then ca.pem should include worker-ca.pem
+	// to let TLS bootstrapped workers acces APIServer.
+	symlinks := []struct {
+		from string
+		to   string
+	}{
+		{"ca.pem", "worker-ca.pem"},
+		{"ca.pem", "etcd-trusted-ca.pem"},
+		{"ca-key.pem", "worker-ca-key.pem"},
```

Comment:
`worker-ca-key.pem` becomes a symlink to an inexistent file when `ca-key.pem` was not written because the user provided `--ca-key-path` while the TLS bootstrapping is disabled.
We should not create the `worker-ca-key.pem` symlink or try to read it afterwards when `includeCAKey` is `false`, the same as what we do for `ca-key.pem` in #877